rust/experimental/lang/src/statestack.rs

```diff
@@ -31,7 +30,8 @@ struct Entry<T> {
 }
 
 pub trait NewState<T> {
```

Comment:
if you have context for this, adding a doc comment would probably benefit future archaeologists (I have no idea what's going on here, for instance)

Reply:
This used to be used to allow the `Context` to either store a `Style` or some debug info... I'm not sure it really _should_ be around anymore... I'm pretty sure I can just replace the `HolderNewState` usage with a top of stack query.